softlayer/resource_softlayer_user.go

```diff
@@ -481,7 +485,13 @@ func resourceSoftLayerUserExists(d *schema.ResourceData, meta interface{}) (bool
 		return false, nil
 	}
 
-	return result.Id != nil && *result.Id == id && err == nil, nil
+	if err != nil {
+		if apiErr, ok := err.(sl.Error); ok && apiErr.StatusCode == 404 {
+			return false, nil
+		}
+		return false, fmt.Errorf("Error retrieving user information: %s", err)
+	}
+	return true, nil
```

Comment:
as a precaution, you should still make sure you have a valid Id in the result and it matches the expected id.

`return result.Id != nil && *result.Id == id, nil`